library/core/src/char/methods.rs

```diff
@@ -336,15 +336,14 @@ impl char {
         // the `radix` is constant and 10 or smaller
         let val = if likely(radix <= 10) {
             // If not a digit, a number greater than radix will be created.
-            (self as u32).wrapping_sub('0' as u32)
+            (self as u8).wrapping_sub(b'0')
```

Comment:
This is wrong. `('\u{0130}' as u8).wrapping_sub(b'0')` = `('0' as u8).wrapping_sub(b'0')`